control-plane-operator/controllers/hostedcontrolplane/oapi/deployment_test.go

```diff
@@ -0,0 +1,235 @@
+package oapi
+
+import (
+	"testing"
+
+	. "github.com/onsi/gomega"
+	configv1 "github.com/openshift/api/config/v1"
+	hyperv1 "github.com/openshift/hypershift/api/hypershift/v1beta1"
+	"github.com/openshift/hypershift/control-plane-operator/controllers/hostedcontrolplane/manifests"
+	"github.com/openshift/hypershift/support/config"
+	"github.com/openshift/hypershift/support/util"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// Ensure certain deployment fields do not get set
+func TestReconcileOpenshiftAPIServerDeploymentNoChanges(t *testing.T) {
+
+	imageName := "oapiImage"
+	// Setup expected values that are universal
+
+	// Setup hypershift hosted control plane.
+	targetNamespace := "test"
+	oapiDeployment := manifests.OpenShiftAPIServerDeployment(targetNamespace)
+	hcp := &hyperv1.HostedControlPlane{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "hcp",
+			Namespace: targetNamespace,
+		},
+	}
+	hcp.Name = "name"
+	hcp.Namespace = "namespace"
+	ownerRef := config.OwnerRefFrom(hcp)
+	serviceServingCA := manifests.ServiceServingCA(hcp.Namespace)
+
+	testCases := []struct {
+		cm                    corev1.ConfigMap
+		auditConfig           *corev1.ConfigMap
+		deploymentConfig      config.DeploymentConfig
+		additionalTrustBundle *corev1.LocalObjectReference
+		clusterConf           *hyperv1.ClusterConfiguration
+	}{
+		// empty deployment config
```

Comment:
can we have a Name field with the test case name?

Reply:
We can, but this is an automatic cherry-pick. I prefer to modify in main and backport it properly